onos-classic/templates/nodeports.yaml

```diff
@@ -1,45 +1,51 @@
 ---
-{{- if .Values.onosOfPort }}
+  {{- if .Values.onosApiPort }}
 apiVersion: v1
 kind: Service
 metadata:
-  name: onos-of
+  name: onos-api
 spec:
   type: NodePort
   ports:
-  - name: onos-of
-    port: 6653
-    nodePort: {{ .Values.onosOfPort }}
+    - name: onos-api
+      port: 8181
+      nodePort: {{ .Values.onosApiPort }}
   selector:
     app: {{ template "fullname" . }}
-{{- end}}
+  {{- end}}
 ---
-{{- if .Values.onosApiPort }}
+  {{- if .Values.onosSshPort }}
 apiVersion: v1
 kind: Service
 metadata:
-  name: onos-api
+  name: onos-ssh
 spec:
   type: NodePort
   ports:
-  - name: onos-api
-    port: 8181
-    nodePort: {{ .Values.onosApiPort }}
+    - name: onos-ssh
+      port: 8101
+      nodePort: {{ .Values.onosSshPort }}
   selector:
     app: {{ template "fullname" . }}
-{{- end}}
+  {{- end}}
+
+# workaround for . not working, see
+# https://github.com/helm/helm/issues/1311
```

Comment:
It's not really even that `.` doesn't work. It actually does work as intended. This is just a limitation of variable scopes in Go templates. It does make for some pretty messy workarounds. We probably won't get anything out of tracking the issue...

Reply:
Agreed, but since it took me a little to understand what was going on I still think it's worth tracking, also to show people why I have done something like that and to help if in the future something similar.

Reply:
Yep it took me a while to understand how it worked too. It can be pretty frustrating.